client/state/comments/actions.js

```diff
@@ -154,19 +155,27 @@ export const deleteComment = (
 	commentId,
 	options = { showSuccessNotice: true },
 	refreshCommentListQuery = null
-) => ( {
-	type: COMMENTS_DELETE,
-	siteId,
-	postId,
-	commentId,
-	options,
-	refreshCommentListQuery,
-	meta: {
-		dataLayer: {
-			trackRequest: true,
+) => ( dispatch, getState ) => {
+	const siteComment = getSiteComment( getState(), siteId, commentId );
+	const previousStatus = siteComment && siteComment.status;
+
+	dispatch( {
+		type: COMMENTS_DELETE,
+		siteId,
+		postId,
+		commentId,
+		options,
+		refreshCommentListQuery,
+		meta: {
+			comment: {
```

Comment:
You are right. I didn't find any other place where `previousStatus` would be set so I thought it should be fine. Calling @jsnajdr as he might be able to shed some more light.

Reply:
@jsnajdr thanks for your explanation!

I may miss something here but one question I have is wouldn't it be more reliable to _calculate_ the counts in a selector based on the comments in the state rather than relying on logic to adjust the count depending on certain actions?

Reply:
I'm not very familiar with the comments code, but I guess that not all comments need to be loaded locally in the state. The comment counts are retrieved from a [dedicated REST endpoint](https://github.com/Automattic/wp-calypso/blob/master/client/state/data-layer/wpcom/sites/comment-counts/index.js#L23).

Reply:
I see, that would make it hard to calculate the count from the local state. Thanks!